### PR TITLE
Multipopulation support for compartment reports

### PR DESCRIPTION
**Pull request by jorblancoa**
- All ranks participate writing all populations

### CODE REVIEW

**Comment by jorblancoa** [comment]
Launched gitlab CI with this branch:
https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/36782

File: src/io/hdf5_writer.cpp [hunk 1]
```diff
@@ -180,10 +176,10 @@ void HDF5Writer::close() {
     if (spikes_attr_type_) {
         H5Tclose(spikes_attr_type_);
         spikes_attr_type_ = 0;
-    }
-    if (file_) {
-        H5Fclose(file_);
-        file_ = 0;
+        if (file_) {
```
Comment (ferdonline) [inline src/io/hdf5_writer.cpp:179]: At a glance, shouldn't all ranks close the file, independently of this `spikes_attr_type_` ?

File: src/library/implementation_interface.hpp [hunk 2]
```diff
@@ -299,12 +345,19 @@ struct SerialImplementation {
         return 0;
     };
     static void close(){};
-    static std::tuple<hid_t, hid_t, hid_t, std::string> prepare_write(
-        const std::string& report_name) {
-        return std::make_tuple(H5Pcreate(H5P_FILE_ACCESS),
-                               H5Pcreate(H5P_DATASET_XFER),
-                               H5Pcreate(H5P_DATASET_XFER),
-                               report_name + FILE_EXTENSION);
+    static void add_communicator(const std::string& /*comm_nam*/){};
```
Comment (ferdonline) [inline src/library/implementation_interface.hpp:348]: nitpick. We could have some docs for each public interface function, at least explaining what it's supposed to do and to help splitting the code.

**Review by ferdonline — APPROVED** [review]

File: src/library/implementation_interface.hpp (hunk 2 above)
Comment (jorblancoa) [reply src/library/implementation_interface.hpp:348]: Totally agree. Will add this to the TODO and will do it in the next PR when I will improve other things also.

**Pull request merged**